darwin/future/core/types/common.py

```diff
@@ -73,18 +80,31 @@ class QueryString:
         Returns a string representation of the QueryString object, in the format "?key1=value1&key2=value2".
     """
 
-    value: Dict[str, str]
+    value: dict[str, list[str] | str]
 
-    def dict_check(self, value: UnknownType) -> Dict[str, str]:
-        assert isinstance(value, dict)
-        assert all(isinstance(k, str) and isinstance(v, str) for k, v in value.items())
-        return value
+    def dict_check(
+        self, value: Mapping[str, list[Stringable] | Stringable]
+    ) -> dict[str, list[str] | str]:
+        mapped: dict[str, list[str] | str] = {}
+        for k, v in value.items():
+            if isinstance(v, list):
+                mapped[k] = [str(x) for x in v]
+            else:
+                mapped[k] = str(v)
+        return mapped
 
-    def __init__(self, value: Dict[str, str]) -> None:
+    def __init__(self, value: Mapping[str, list[Stringable] | Stringable]) -> None:
```

Comment:
Mapping here is a more free-form Dict. As per the mypy docs, it's better to use the constructs like Mapping/Sequence as input parameters, and more concrete ones like dict/list as return types.